src/modules/displaymanager/main.py

```diff
@@ -633,41 +636,57 @@ def desktop_environment_setup(self, default_desktop_environment):
                 )
             )
 
-    def greeter_setup(self):
-        lightdm_conf_path = os.path.join(
-            self.root_mount_point, "etc/lightdm/lightdm.conf"
-            )
+    def find_preferred_greeter(self, greeters_dir):
+        """
+        On Debian, lightdm-greeter.desktop is typically a symlink managed
+        by update-alternatives pointing to /etc/alternatives/lightdm-greeter
+        which is also a symlink to a real .desktop file back in /usr/share/xgreeters/
 
-        # configure lightdm-greeter
-        greeter_path = os.path.join(
-            self.root_mount_point, "usr/share/xgreeters"
-            )
+        Returns a path *into the mounted target* of the preferred greeter -- usually
+        a .desktop file that specifies where the actual executable is. May return
+        None to indicate nothing-was-found.
+        """
+        greeters_dir = "usr/share/xgreeters"
+        greeters_target_path = os.path.join(self.root_mount_point, greeters_dir)
+        available_names = os.listdir(greeters_target_path)
+        available_names.sort()
+        desktop_names = [n for n in self.preferred_greeters if n in available_names] # Preferred ones
+        if desktop_names:
+            return desktop_names[0]
+        desktop_names = [n for n in available_names if n.endswith(".desktop")] # .. otherwise any .desktop
+        if desktop_names:
+            return desktop_names[0]
+        desktop_names = [n for n in available_names if not n.startswith(".")] # .. otherwise any non-dot-file
+        if desktop_names:
+            return desktop_names[0]
+        if available_names: # Anything?
+            return available_names[0]
```

Comment:
Are these conditions valid?  Should we just be throwing an exception or returning an empty array here?

Reply:
Good point, the last one doesn't make any sense (the one on L660 does, if none of the preferred ones is matched).

Reply:
Isn't the one at line 659 also invalid?  That looks like it accepts any file that doesn't start with `.`